### PR TITLE
Extend plugin paths

### PR DESCRIPTION
**Pull request by Nicogene**
This PR aims to extend the plugin search directories, gazebo-fashioned.

TODO:

- [ ] Check old functionality(local dir)
- [ ] Check new functionality(plugin paths)
- [ ] Add documentation

### CODE REVIEW

**Comment by traversaro** [comment]
FYI the sharedlibpp version as modified in blockfactory also contains some of this logic, see https://github.com/robotology/blockfactory/pull/32 .

**Comment by Nicogene** [comment]
I would start with a quick and dirty solution and see if works, and then the introduction of `sharedlipp`would be nice, but unfortunately we have to do yet another fork :disappointed: 

The best solution would be having a unique `sharedlibpp` and use it everywhere (https://github.com/icub-tech-iit/tickets/issues/379).

**Comment by traversaro** [comment]
> I would start with a quick and dirty solution and see if works, and then the introduction of sharedlippwould be nice, but unfortunately we have to do yet another fork 😞

Yes, that was just a link that I shared FYI, going forward with this change is definitely the way to go for now.

File: src/blocktestrunner/libraryLoader.h [hunk 1]
```diff
@@ -37,12 +37,14 @@ class LibraryLoader
         void stop();
 
     private:
+        std::vector<std::string> pluginPaths;
```
Comment (triccyx) [inline src/blocktestrunner/libraryLoader.h]: Plase use _ at the end of the variable name

**Review by triccyx — CHANGES_REQUESTED** [review]

**Review by triccyx — APPROVED** [review]

**Pull request merged**